plugin-server/src/utils/retries.ts

```diff
@@ -2,8 +2,35 @@ import { RetryError } from '@posthog/plugin-scaffold'
 
 import { runInTransaction } from '../sentry'
 import { Hub } from '../types'
+import { status } from '../utils/status'
 import { AppMetricIdentifier, ErrorWithContext } from '../worker/ingestion/app-metrics'
 
+// Simple retries in our code
+export const defaultRetryConfig = {
+    // for easy value changes in tests
+    RETRY_INTERVAL_DEFAULT: 1000,
+    MAX_RETRIES_DEFAULT: 3,
+}
+
+export async function promiseRetry<T>(
+    fn: () => Promise<T>,
+    name: string,
+    retries = defaultRetryConfig.MAX_RETRIES_DEFAULT,
```

Comment:
Pretty sure it's good, I get paranoid when I see defaults having been burned by python, but I _think_ these default value expressions are evaluated at function call rather than at definition time 👍

Reply:
yup this is fine per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters